src/dev/api.ts

```diff
@@ -197,6 +220,29 @@ export function apiRoutes(
       // TODO: Handle errors
       if (!response) throw new Error('Failed to fetch frame')
 
+      const cloned = response.clone()
+
+      if (!response.ok) {
+        const message = await responseToBaseErrorMessage(cloned)
```

Comment:
This block repeats twice in this file. Maybe it can be abstracted?

Reply:
I think it's ok for now.

Mine rule of thumb is:
- Code block used once: ok;
- Code block used twice: be cautious;
- Code block used 3 times: recompose.

In this case i'm a bit afraid that decomposing this code block might break inferred return type definitions that are later used in other locations to provide a typed `Response.json()` returns.